src/domain/user.js

```diff
@@ -209,4 +213,22 @@ export default class User {
 
     return foundUsers.map((user) => User.fromDb(user))
   }
+
+  static async _findManyThroughUser(key, value) {
+    const query = {
+      include: {
+        profile: true
+      }
+    }
+
+    if (key !== undefined && value !== undefined) {
+      query.where = {
+        [key]: value
+      }
+    }
```

Comment:
Nice. This could perhaps be extracted out to its own query builder function and improved later